api/client.go

```diff
@@ -363,18 +363,18 @@ func (s *Status) Task(name string, q *QueryParam) (map[string]TaskStatus, error)
 	return taskStatuses, nil
 }
 
-// Task can be used to query the task endpoints
-type Task struct {
+// TaskClient can be used to query the task endpoints
```

Comment:
it collides with the Task rep, also I thought TaskClient made the struct clearer. 

Also, we could make the c *Client anonymous, then you wouldn't have a stutter?

Reply:
I decided to refactor Status and Task client to be TaskClient and StatusClient respectively. The `c *Client` field has now been made anonymous so that TaskClient and StatusClient now inherit *Client

I feel that:
- This will eliminate naming collisions with representations in the future
- The names now better reflect what these structs are